collect_app/src/main/java/org/odk/collect/android/widgets/IBinaryNameWidget.java

```diff
@@ -0,0 +1,8 @@
+package org.odk.collect.android.widgets;
+
+/**
+ * @author James Knight
+ */
+public interface IBinaryNameWidget extends IBinaryWidget {
```

Comment:
Yeah, I wasn't really sure what to call it either. In general, the "IBinaryWidgets" and what qualifies as "binary data" within them are rather loose. These were a subset of those that had a `String binaryName;` field, and an underlying `deleteMedia()` method that got called during `clearBinaryData()`.

I like your BinaryWidgetWithMediaFile, or maybe even just FileWidget?

Reply:
`FileWidget` seems good! And I think `binaryName` is really `fileName`, right? I think that where names obscure concepts, you can feel free to change them.

Reply:
Yeah, I wanted to avoid making too many sweeping changes in this one that weren't directly related to improving testability or getting rid of existing warnings. I'll take another pass through these changed files and see if there's any just general cleanup I can do before moving on to the rest of the Widgets.

Reply:
Perfection!!!! Or rethinking all naming as you rework the hierarchies would be great, too.